### ISSUE TITLE
Cleanup after merging Serialization PR in dev/api

### ISSUE THREAD

**Issue by joperezr** [dotnet/corefx#10714]
PR #10667 added Serialization back to Core. It came with a lot of ToDo's so this issue is to track those and make sure they get done.They are:

- [ ] Use SR instead of hard-coded strings once it is available in System.Runtime.
- [ ] Uncomment Debug.Asserts once it is available in System.Runtime
- [ ] Remove P2P references once corefx versions are updated.
- [ ] Remove [temporary workaround](https://github.com/dotnet/corefx/pull/10667/files#diff-340897543d74ba6c30ea455b13aace92R1628) in ObjectManager.cs

**Comment by AlexGhiondea** [comment]
@joperezr is this something we can start working on now?

**Comment by joperezr** [comment]
yup, I was planning on taking this on once I'm done with investigating the test builds.

**Comment by shmao** [comment]
@joperezr Has this been done? Can we close this issue?

**Comment by joperezr** [comment]
the only thing that couldn't be done yet is to remove the workaround in ObjectManager.cs. We couldn't find an easy and clean way for removing those extension methods from there without having to expose them to the public on coreclr, so that is why this issue is not yet fully done, but I'm not sure it can actually be done without a major refactoring so we might end up keeping it as is.

**Comment by danmosemsft** [comment]
-> future as not necesasry for 2.0

**Comment by karelz** [comment]
@joperezr @zhenlan @shmao can we decide if it is something we want to do eventually, or just close it?
It should not be assigned to @joperezr as he is not working on it ...

**Comment by zhenlan** [comment]
@karelz, if this issue is about `System.Runtime.Serialization.Formatters`, can you please move it to proper team?

**Comment by joperezr** [comment]
@stephentoub what do you think? do we want to eventually fix the issue with ObjectManager or should we just close the issue?

**Comment by stephentoub** [comment]
I think we should fix it.  In general I think we should aim to fix anywhere we're using reflection from one library to another; such reflection will wreak havoc on tools that rely on being able to see the dependencies.

**Comment by karelz** [comment]
@zhenlan I didn't see any evidence in the description that it is about `System.Runtime.Serialization.Formatters`, that's why I put it here - in future, please feel free to change areas when you have more info.
I'll update the title and change area for this particular issue ...
